manual_tests/internal/manager.go

```diff
@@ -51,7 +59,17 @@ func (m *Manager) SelectStorageType(storageType fixtures.StorageType) error {
 }
 
 func (m *Manager) WalletForUser(user fixtures.UserConfig) (sdk.Interface, error) {
-	userWallet, err := wallet.New(m.config.BSVNetwork, user.PrivKey, m.storageInfra.Provider)
+	var storageProvider wdk.WalletStorageProvider
+
+	if m.remoteStorage != nil {
+		storageProvider = m.remoteStorage
+	} else if m.storageInfra != nil {
+		storageProvider = m.storageInfra.Provider
+	} else {
+		return nil, fmt.Errorf("no storage provider configured")
+	}
```

Comment:
```suggestion
switch {
	case m.remoteStorage != nil:
		storageProvider = m.remoteStorage
	case m.storageInfra != nil:
		storageProvider = m.storageInfra.Provider
	default:
		return nil, fmt.Errorf("no storage provider configured")
	}
```
or you could make helper function 

```go
func (m *Manager) resolveStorageProvider() (wdk.WalletStorageProvider, error) {
	if m.remoteStorage != nil {
		return m.remoteStorage, nil
	}
	if m.storageInfra != nil {
		return m.storageInfra.Provider, nil
	}
	return nil, fmt.Errorf("no storage provider configured")
}

func (m *Manager) WalletForUser(user fixtures.UserConfig) (sdk.Interface, error) {
	storageProvider, err := m.resolveStorageProvider()
	if err != nil {
		return nil, err
	}

	userWallet, err := wallet.New(m.config.BSVNetwork, user.PrivateKey(), storageProvider)
	if err != nil {
		return nil, fmt.Errorf("failed to create wallet for user %s: %w", user.Name, err)
	}
	return userWallet, nil
}
```

Reply:
changed